### ISSUE TITLE
Issue with gufi_ls

### ISSUE THREAD

**Issue by swapng** [mar-file-system/GUFI#136]
Special chars are not handled in gufi_ls cmd, e.g. if my directory name containing special char like (dir#1@$) then gufi_ls is failing and not giving expected output.
I also verified by creating dirs having special chars. but gufi_ls is failing only dir. which are having '#' with their name.

**Comment by calccrypto** [comment]
Hi. Can you please provide a minimal, reproducible example? I am getting output I expect from a small tree:

```bash
mkdir root
mkdir root/dir#1@$
touch root/dir#1@$/file
gufi_dir2index root index

gufi_ls
gufi_ls root
gufi_ls root/dir#1@$
gufi_ls root/dir#1@$/file
gufi_ls -R
```

**Comment by swapng** [comment]
Yes With current version of GUFI it's working for me as well, but i'm using older version of GUFI [FROM THIS COMMIT](https://github.com/mar-file-system/GUFI/tree/1b320b18c48ad42d076a078d2faaa61d448e9806) COMMIT ID--> `1b320b18c48ad42d076a078d2faaa61d448e9806`

If possible could you please provide the commit from which this issue gets resolved so that I could take a look.

Thank you!

**Comment by calccrypto** [comment]
The commit that fixed special character handling was 062375b6b8e7259e23b533d254ac34dcd321399d. Testing was added in the commit after 4a366b23875636ccca96e0a0ba92d3fde99dfd11. An additional fix was added in df02d97bcf6826cde381a256eac80d7233bc0ecb.

**Comment by swapng** [comment]
Thanks!
